core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -0,0 +1,388 @@
+    this.warehouseLocation = warehouse.replaceAll("/$", "");
+
+    if (name != null) {
+      this.catalogName = name;
+    }
+
+    String fileIOImpl = properties.get(CatalogProperties.FILE_IO_IMPL);
+    this.io = fileIOImpl == null ? new HadoopFileIO(conf) : CatalogUtil.loadFileIO(fileIOImpl, properties, conf);
+
+    try {
+      LOG.debug("Connecting to Jdbc database {}", properties.get(CatalogProperties.URI));
+      connections = new JdbcClientPool(uri, properties);
+      initializeCatalogTables();
+    } catch (SQLTimeoutException e) {
+      throw new UncheckedSQLException(e, "Cannot initialize jdbc catalog: Query timed out");
+    } catch (SQLTransientConnectionException | SQLNonTransientConnectionException e) {
+      throw new UncheckedSQLException(e, "Cannot initialize jdbc catalog: Connection failed");
+    } catch (SQLException e) {
+      throw new UncheckedSQLException(e, "Cannot initialize jdbc catalog");
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to initialize", e);
+    }
+  }
+
+  private void initializeCatalogTables() throws InterruptedException, SQLException {
+    LOG.trace("Creating database tables (if missing) to store iceberg catalog");
+    connections.run(conn -> {
+      DatabaseMetaData dbMeta = conn.getMetaData();
+      ResultSet tableExists = dbMeta.getTables(null, null, JdbcUtil.CATALOG_TABLE_NAME, null);
+
+      if (tableExists.next()) {
+        return true;
+      }
+
+      LOG.debug("Creating table {} to store iceberg catalog!", JdbcUtil.CATALOG_TABLE_NAME);
+      return conn.prepareStatement(JdbcUtil.CREATE_CATALOG_TABLE).execute();
+    });
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    return new JdbcTableOperations(connections, io, catalogName, tableIdentifier);
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier table) {
+    return SLASH.join(defaultNamespaceLocation(table.namespace()), table.name());
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+
+    int deletedRecords;
+    try {
+      deletedRecords = connections.run(conn -> {
+        try (PreparedStatement sql = conn.prepareStatement(JdbcUtil.DROP_TABLE_SQL)) {
+          sql.setString(1, catalogName);
+          sql.setString(2, JdbcUtil.namespaceToString(identifier.namespace()));
+          sql.setString(3, identifier.name());
+          return sql.executeUpdate();
+        }
+      });
+    } catch (SQLException e) {
+      throw new UncheckedSQLException(e, "Failed to drop %s", identifier);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to dropTable", e);
+    }
+
+    if (deletedRecords > 0) {
+      LOG.info("Successfully dropped table {}.", identifier);
```

Comment:
I think we should base logging on what's done in the `HiveCatalog`. That way it is consistent. Here's the log for drop table: https://github.com/apache/iceberg/blob/master/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java#L175

That also doesn't log if no table was dropped because it doesn't exist. I think that's the right call because we don't want to overwhelm the log with what didn't happen. At most that information should be logged at debug.

Reply:
aligned logs with hive, 

it seems when table not exists hive [throws exception](https://github.com/apache/iceberg/blob/15616941461aaad3f03568ce4880171b86a06547/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java#L178) then its handled in catch section, on jdbc case there is no exception thrown but deletedRecords=0 , what do you think changing loglevel to debug and keeping `retun false` ?



```
} catch (NoSuchTableException | NoSuchObjectException e) {
LOG.info("Skipping drop, table does not exist: {}", identifier, e);
<span class="pl-k">return</span> <span class="pl-c1">false</span>;
```

Reply:
I'm okay with either debug or info for this one.